src/garage/tf/models/cnn_mlp_merge_model.py

```diff
@@ -0,0 +1,162 @@
+                 filter_dims,
+                 num_filters,
+                 strides,
+                 hidden_sizes=(256, ),
+                 output_dim=1,
+                 action_merge_layer=-2,
+                 name=None,
+                 padding='SAME',
+                 max_pooling=False,
+                 pool_strides=(2, 2),
+                 pool_shapes=(2, 2),
+                 cnn_hidden_nonlinearity=tf.nn.relu,
+                 cnn_hidden_w_init=tf.initializers.glorot_uniform(),
+                 cnn_hidden_b_init=tf.zeros_initializer(),
+                 hidden_nonlinearity=tf.nn.relu,
+                 hidden_w_init=tf.initializers.glorot_uniform(),
+                 hidden_b_init=tf.zeros_initializer(),
+                 output_nonlinearity=None,
+                 output_w_init=tf.initializers.glorot_uniform(),
+                 output_b_init=tf.zeros_initializer(),
+                 layer_normalization=False):
+        super().__init__(name)
+
+        if not max_pooling:
+            self.cnn_model = CNNModel(
+                filter_dims=filter_dims,
+                num_filters=num_filters,
+                hidden_w_init=cnn_hidden_w_init,
+                hidden_b_init=cnn_hidden_b_init,
+                strides=strides,
+                padding=padding,
+                hidden_nonlinearity=cnn_hidden_nonlinearity)
+        else:
+            self.cnn_model = CNNModelWithMaxPooling(
+                filter_dims=filter_dims,
+                num_filters=num_filters,
+                hidden_w_init=cnn_hidden_w_init,
+                hidden_b_init=cnn_hidden_b_init,
+                strides=strides,
+                padding=padding,
+                pool_strides=pool_strides,
+                pool_shapes=pool_shapes,
+                hidden_nonlinearity=cnn_hidden_nonlinearity)
+
+        self.mlp_merge_model = MLPMergeModel(
+            output_dim=output_dim,
+            hidden_sizes=hidden_sizes,
+            concat_layer=action_merge_layer,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            layer_normalization=layer_normalization)
+
+    def network_input_spec(self):
+        """Network input spec.
+
+        Return:
+            list[str]: List of key(str) for the network inputs.
+        """
+        return ['state', 'action']
+
+    # pylint: disable=arguments-differ
+    def _build(self, state, action, name=None):
+        """Build the model and return the outputs.
+
+        This builds the model such that the output of the CNN is fed
+        to the MLP. The CNN receives the state as the input. The MLP
+        receives two inputs, the output of the CNN and the action
+        tensor.
+
+        Args:
+            state (tf.Tensor): State placeholder tensor.
```

Comment:
please provide the shapes of input/output tensors. See the style for doing this in `CONTRIBUTING.md` and `src/garage/_dtypes.py`

Reply:
fixed